Mention Kotlin DSL faster first use in release notes (#26631)
gradle/gradle: subprojects/docs/src/docs/release/notes.md

Gradle's [Kotlin DSL](userguide/kotlin_dsl.html) provides an enhanced editing experience in supported IDEs compared to the traditional Groovy DSL — auto-completion, smart content assist, quick access to documentation, navigation to source, and context-aware refactoring.

#### Faster first use

> With an empty [Gradle User Home](userguide/directory_layout.html#dir:gradle_user_home), before compiling the first `.gradle.kts` script, Gradle generates the Gradle API Kotlin DSL Extensions JAR.

Reviewer: Feels a bit "abrupt", like it needs a gentler introduction, more context, not exactly sure what.

Something like "when running a build with an empty Gradle user home (for example first install, ephemeral CI, whatnot)", before we compile ....

Take the next sentence and put it first maybe?

Author: Agreed. Does it feels better now with Laura's suggestions applied?

Reviewer: Definitely.

> With an empty [Gradle User Home](userguide/directory_layout.html#dir:gradle_user_home), before compiling the first `.gradle.kts` script, Gradle generates the Gradle API Kotlin DSL Extensions JAR.

Reviewer 2: ```suggestion
Gradle generates a Gradle API Kotlin DSL Extensions JAR before compiling the first `.gradle.kts` script when the [Gradle User Home](userguide/directory_layout.html#dir:gradle_user_home) is empty.
```

> This happens on every developer machine or CI agent, often with a cold [Gradle daemon](userguide/gradle_daemon.html).

Reviewer: The "often with a cold daemon" statement seems misleading. This is not about the daemon being cold, but about the gradle user home not having stuff.

Author: It adds up. 99% of the time you'll have both a cold daemon and empty user home, making this even slower.

> This happens on every developer machine or CI agent, most often with a cold [Gradle daemon](userguide/gradle_daemon.html).

Reviewer 2: ```suggestion
This happens on every developer machine or CI agent, often with a cold [Gradle daemon](userguide/gradle_daemon.html).
```

> This is quite a resource intensive process which accounts for around 4 seconds on beefy machines ; much more on slower machines.

Reviewer 2: ```suggestion
This resource-intensive process takes around 4 seconds on powerful machines and is much longer on slower machines.
```

> Starting with this release, the Kotlin extensions for the Gradle API are now shipped at part of the Gradle distribution.

Reviewer: ```suggestion
Starting with this release, the Kotlin extensions for the Gradle API are now shipped as part of the Gradle distribution.
```

> Starting with this release, the Kotlin extensions for the Gradle API are now shipped at part of the Gradle distribution.

Reviewer 2: ```suggestion
The Kotlin extensions for the Gradle API now ship as part of the Gradle distribution.
```

Author: :+1: incorporated manually given conflicts

> The consequence is that first-use of a Gradle version for compiling `.gradle.kts` scripts is faster.

Reviewer 2: ```suggestion
Consequently, the first use of a Gradle version for compiling `.gradle.kts` scripts is much faster.
```
